### PR TITLE
docs: Improves README.md file

### LINKED ISSUE #293: Improve README.md
- Improve the `README.md`.
- Mention all the required information of the repository.
- Also mentions all the important commands.

### PR DESCRIPTION
**Pull request by ankur0904**
Fix #293
This pull request introduces enhancements to the `README.md` file for the reflex-web repository-
- provide how to setup the project locally,
- provide comprehensive and clear instructions for setting up the project with required `commands`
- visual improvements have been made to the README
- addition of badges.

**Question**
- Can I add `reflex` [logo](https://github.com/reflex-dev/reflex-web/blob/main/assets/Reflex.svg) to the `README.md` ?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,90 @@
...
+![versions](https://img.shields.io/pypi/pyversions/reflex.svg)
+[![Documentation](https://img.shields.io/badge/Documentation%20-Introduction%20-%20%23007ec6)](https://reflex.dev/docs/getting-started/introduction)
+[![Discord](https://img.shields.io/discord/1029853095527727165?color=%237289da&label=Discord)](https://discord.gg/T5WSbC2YtQ)
+</div>
+
+# Overview
+
+A public repository which contains the source code of the reflex.dev website.
+
+- It's built with [Reflex](https://reflex.dev/), a Python web framework.
```
Comment (picklelo) [inline README.md:19]: instead of "a Python web framework" can we make it "a framework to build web apps in pure Python"

File: README.md [hunk 2]
```diff
@@ -1,2 +1,90 @@
...
+- You can find everything related to Reflex at [reflex.dev](https://reflex.dev/).
+
+## About Reflex
+
+Reflex.dev is a groundbreaking web development framework that empowers Python developers to seamlessly create web apps without the nuances of traditional frontend tools. Its open-source core framework ensures flexibility for projects of any scale. Explore the future of web development at [reflex.dev](https://reflex.dev/).
```
Comment (picklelo) [inline README.md:24]: For the first sentence I think we don't need to say "groundbreaking" let's just make it something like "Reflex is a full-stack web framework that allows developers to build their app in pure Python, and deploy with a single command".

**Review by picklelo — CHANGES_REQUESTED** [review]
Thanks for adding this, just a couple comments

**Comment by ankur0904** [comment]
@picklelo Made the suggested changes.

**Review by picklelo — APPROVED** [review]
looks good, thanks for adding!

**Pull request merged**